testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/TestHiveStorageFormats.java

```diff
@@ -240,6 +241,7 @@
         return Stream.of(storageFormats())
                 // everything but Avro supports nanoseconds
                 .filter(format -> !"AVRO".equals(format.getName()))
+                .filter(format -> !format.getSessionProperties().containsKey("hive.experimental_parquet_optimized_writer_enabled"))
```

Comment:
- The comment above says that all but Avro support nanoseconds, but new parquet writer clearly does not. Update the comment.

- The check here should also check that a) name = "PARQUET" and b) session property value is "true".

- Please file an issue for this, and add TODO comment here, with a link to the issue.

Reply:
@findepi Would it make sense to have the issue: https://github.com/trinodb/trino/issues/5357 also responsible for updating this test case (skipping nanoseconds for optimized reader) to verify correctness? Either way, I will create tag it in the code.

Reply:
https://github.com/trinodb/trino/issues/8715

Reply:
Also, which extra session property are you referring to?